### PR TITLE
Update the documentation and the push process

### PR DESCRIPTION
**Pull request by ZPascal**

### CODE REVIEW

**Comment by ZPascal** [comment]
Hi @jackton1, could you please take a look?

File: README.md [hunk 1]
```diff
@@ -39,6 +39,28 @@ jobs:
         branch: ${{ github.ref }}
 ```
 
+An example workflow to use the branch parameter to push the changes to specified branch e.g. an PR related branch:
```
Comment (jackton1) [inline README.md:42]:
```suggestion
An example workflow to use the branch parameter to push the changes to a specified branch e.g. a Pull Request branch:
```

File: README.md [hunk 2]
```diff
@@ -39,6 +39,28 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+    - name: Commit files
+      run: |
+        git config --local user.email "github-actions[bot]@users.noreply.github.com"
+        git config --local user.name "github-actions[bot]"
+        git commit -m "Add changes" -a
```
Comment (jackton1) [inline README.md:57]:
```suggestion
        git commit -a -m "Add changes"
```
I suggest adding options that don't take any argument before the ones that do.

File: README.md [hunk 3]
```diff
@@ -39,6 +39,28 @@ jobs:
         branch: ${{ github.ref }}
 ```
 
+An example workflow to use the branch parameter to push the changes to specified branch e.g. an PR related branch:
+
+```yaml
+jobs:
```
Comment (jackton1) [inline README.md:45]: Can we add the event that would likely be used to trigger this action as `head_ref` is only set for specific events.

```
The head_ref or source branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.
```
See: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

**Review by jackton1 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (ZPascal) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (ZPascal) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (ZPascal) [reply README.md:48]: Done

**Review by jackton1 — APPROVED** [review]

**Pull request merged**